src/eval/semantics.cpp

```diff
@@ -46,43 +46,43 @@ Number Semantics::mod(const Number& a, const Number& b) {
 }
 
 Number Semantics::pow(const Number& base, const Number& exp) {
+  if (base == Number::ONE) {
+    return 1;  // 1^x is always 1
+  }
+  if (base == -1) {
+    return exp.odd() ? -1 : 1;  // (-1)^x
+  }
   if (base == Number::INF || exp == Number::INF) {
     return Number::INF;
   }
+  if (exp < Number::ZERO) {
```

Comment:
this case has different results depending on the value of the base (see also the failing test cases)

Reply:
fixed